### PR TITLE
[#4362] feat(web): Add GitHub stars and forks number for WebUI

### PR DESCRIPTION
**Pull request by SZL741023**
### What changes were proposed in this pull request?

1. Add GitHub image, stars icon and fork icon on the Navbar.
2. Send GET request to GitHub API to fetch the number of stars and forks and show on the Navbar
3. If the number of stars and forks more than 1000, it will show 1.x k.
4. Github image and stars icon can link to Gravitino repo, fork icon can link to Gravitino fork page.
5. Position version text next to Gravitino logo.

### Why are the changes needed?

These change can improve user experience and efficiency.

Fix: #4362 

### Does this PR introduce _any_ user-facing change?

1. Add Github image, stars icon and fork icon.
2. Position version text next to Gravitino logo.

### How was this patch tested?

1. Check the data was fetched from GitHub API.
2. console.log the number more than 1000, manual check it will change to 1.x k.

image

### CODE REVIEW

**Comment by xunliu** [comment]
hi @SZL741023 Thank you for your contributions.
@LauraXia123 Please help review this PR, Thanks.

File: web/web/src/app/rootLayout/GitHubInfo.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const GitHubInfo = ({ stars, forks, username, repository }) => {
+  const githubUrl = `https://github.com/${username}/${repository}`
+  const githubForkUrl = `https://github.com/${username}/${repository}/fork`
+  const githubLogoUrl = (process.env.NEXT_PUBLIC_BASE_PATH ?? '') + '/icons/github-mark.svg'
+  const forkLogoUrl = (process.env.NEXT_PUBLIC_BASE_PATH ?? '') + '/icons/git-fork.svg'
+
+  return (
+    <Box className={'twc-flex  twc-gap-x-3 twc-bg-customs-lightBg twc-px-3 twc-py-2 twc-rounded-full'}>
+      <Link href={githubUrl}>
+        <Image className={'twc-align-middle'} src={githubLogoUrl} width={24} height={24} alt='logo' />
```
Comment (LauraXia123) [inline web/web/src/app/rootLayout/GitHubInfo.js:34]:
```suggestion
        <Image className={'twc-align-middle'} src={githubLogoUrl} overrideSrc={githubLogoUrl} width={24} height={24} alt='logo' />
```

File: web/web/src/app/rootLayout/GitHubInfo.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <Link href={githubForkUrl} className={'twc-no-underline  twc-bg-customs-dark twc-rounded-full '}>
+          <Typography
+            className={
+              'twc-flex twc-items-center twc-gap-2 twc-text-customs-black twc-px-2.5 twc-py-1 twc-text-[0.75rem] twc-font-bold'
+            }
+          >
+            <Image
+              className={'twc-align-middle twc-text-customs-white'}
+              src={forkLogoUrl}
+              width={24}
```
Comment (LauraXia123) [inline web/web/src/app/rootLayout/GitHubInfo.js:46]:
```suggestion
              overrideSrc={forkLogoUrl}
              width={24}
```

File: web/web/src/app/rootLayout/AppBar.js [hunk 3]
```diff
@@ -38,15 +38,25 @@ import {
...
 import { useSearchParams } from 'next/navigation'
 import { useRouter } from 'next/navigation'
 import { useAppSelector, useAppDispatch } from '@/lib/hooks/useStore'
 import { fetchMetalakes } from '@/lib/store/metalakes'
 
 const fonts = Roboto({ subsets: ['latin'], weight: ['400'], display: 'swap' })
+function checkCountMore1k(count) {
+  let newCount = count
+  if (count >= 1000) {
+    newCount = (count / 1000).toFixed(1) + 'k'
```
Comment (LauraXia123) [inline web/web/src/app/rootLayout/AppBar.js:55]: if count over 10000, 100000 or more? I suggest adding a format filter funtiion at `/gravitino/web/web/src/lib/utils/index.js`

File: web/web/src/app/rootLayout/AppBar.js [hunk 4]
```diff
@@ -56,14 +66,31 @@ const AppBar = () => {
...
   const logoSrc = (process.env.NEXT_PUBLIC_BASE_PATH ?? '') + '/icons/gravitino.svg'
+  const username = 'apache'
+  const repository = 'gravitino'
+  const [githubData, setGithubData] = useState({ stars: 0, forks: 0 })
 
   useEffect(() => {
+    const fetchGitHubData = async () => {
+      try {
+        const response = await axios.get(`https://api.github.com/repos/${username}/${repository}`)
+        let { stargazers_count, forks_count } = response.data
```
Comment (LauraXia123) [inline web/web/src/app/rootLayout/AppBar.js:77]: In gravitino, axios have imported at `gravitino/web/web/src/lib/utils/axios/index.js` and export `defHttp`,.
And all the response data are all managed. 
I think the github star data can managed at `/gravitino/web/web/src/lib/store/sys/index.js`

Let me know if you have any questions.

Comment (SZL741023) [reply web/web/src/app/rootLayout/AppBar.js:77]: Hi @LauraXia123,
Thanks for suggestion.

I have some question for you.

1. The format filter function that you mean is like image below?
image

2. How to use defHttp to get GitHub API, I use ```const response = defHttp.get({url: githubAIP})```, but always response AxiosError, what's wrong with my code?
image

3. How can I get the svgs in ```/gravitino/web/web/src/lib/icons/svg/```
4. Why do you suggest me to add overrideSrc, I use this keyword to search information, but I can't find anything. what does this attribute means?

thx

File: web/web/src/app/rootLayout/AppBar.js [hunk 5]
```diff
@@ -64,6 +66,7 @@ const AppBar = () => {
     const metalakeItems = store.metalakes.map(i => i.name)
     setMetalakes(metalakeItems)
     // eslint-disable-next-line react-hooks/exhaustive-deps
+    axios.get('/')
```
Comment (LauraXia123) [inline web/web/src/app/rootLayout/AppBar.js:69]: What is this request for?

File: web/web/src/lib/api/github/index.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+import { defHttp } from '@/lib/utils/axios'
+import { ContentTypeEnum } from '@/lib/enums/httpEnum'
+
+const githubApis = {
+  headers: {
+    'Content-Type': ContentTypeEnum.JSON,
+    Accept: 'application/vnd.github+json'
+  },
```
Comment (LauraXia123) [inline web/web/src/lib/api/github/index.js:27]: The `headers` here can be removed.

File: web/web/src/app/rootLayout/AppBar.js (hunk 5 above)
Comment (SZL741023) [reply web/web/src/app/rootLayout/AppBar.js:69]: For testing axios function, I forget to remove it.

File: web/web/src/lib/api/github/index.js (hunk 6 above)
Comment (SZL741023) [reply web/web/src/lib/api/github/index.js:27]: OK, I will remove it later.
thx

Comment (SZL741023) [reply web/web/src/lib/api/github/index.js]: Done.
Thanks for suggestion.

**Comment by LauraXia123** [comment]
Thank you very much for your contribution. LGTM

**Review by xunliu — APPROVED** [review]
@SZL741023 Thank you for your contributions.

**Review by mchades — APPROVED** [review]

**Pull request merged**